### PR TITLE
Message History:  Request and Reply structs

### PR DESCRIPTION
**Pull request by tuddman**
## Summary

- Introduces `HistoryRequest` and `HistoryReply` structs.
- Moves the initialization of the `sync_group` to be self contained in the `message_history` mod.

### CODE REVIEW

File: xmtp_mls/src/groups/message_history.rs [hunk 1]
```diff
@@ -130,4 +210,13 @@ mod tests {
 
         let _ = amal_b.send_message_history_request().await;
     }
+
+    #[tokio::test]
+    async fn test_prepare_messages_to_send() {
+        let wallet = generate_local_wallet();
+        let amal_a = ClientBuilder::new_test_client(&wallet).await;
+        // let group_a = amal_a.create_group(None);
+        let messages_result = amal_a.prepare_messages_to_send().await;
```
Comment (neekolas) [inline xmtp_mls/src/groups/message_history.rs:219]: This test doesn't do a whole lot. Would be good to send a message and make sure that it gets added to the bundle

**Review by neekolas — APPROVED** [review]

File: xmtp_mls/src/groups/message_history.rs [hunk 2]
```diff
@@ -43,27 +60,78 @@ where
...
+    fn from(req: HistoryRequest) -> Self {
+        MessageHistoryRequest {
+            pin_code: req.pin_code,
+            request_id: req.request_id,
+        }
+    }
+}
+
+struct HistoryReply {
+    request_id: String,
```
Comment (neekolas) [inline xmtp_mls/src/groups/message_history.rs:109]: Doesn't this struct need to have some sort of encryption keys in it?

Comment (tuddman) [reply xmtp_mls/src/groups/message_history.rs:109]: Yep.  I was going to put that change in the proto first, and once that's in, make the corresponding addition here, too.

Comment (neekolas) [reply xmtp_mls/src/groups/message_history.rs:109]: 👍

Comment (tuddman) [reply xmtp_mls/src/groups/message_history.rs:109]: https://github.com/xmtp/proto/pull/179

File: xmtp_mls/src/groups/message_history.rs [hunk 3]
```diff
@@ -43,27 +60,78 @@ where
         };
         Ok(())
     }
+
+    #[allow(dead_code)]
+    pub(crate) async fn prepare_messages_to_send(
```
[the remark this answers is not in the document]
Comment (neekolas) [reply xmtp_mls/src/groups/message_history.rs:65]: Maybe `prepare_messages_to_sync`

File: xmtp_mls/src/groups/message_history.rs (hunk 1 above)
Comment (tuddman) [reply xmtp_mls/src/groups/message_history.rs]: The next PR will actually be the roundtrip-ing.. I'll for sure be adding a number of tests around that.  This test is really a stub for that, at this point.

**Pull request merged**